plugins/ril.c

```diff
@@ -345,19 +352,26 @@ static void ril_connected(struct ril_msg *message, gpointer user_data)
 	/* TODO: need a disconnect function to restart things! */
 	rd->connected = TRUE;
 
-	DBG("calling set_powered(TRUE)");
+	DBG("calling set_powered(FALSE) on connected");
 
-	ofono_modem_set_powered(modem, TRUE);
+	g_ril_request_power(rd->ril, FALSE, &rilp);
+
+	if (g_ril_send(rd->ril, RIL_REQUEST_RADIO_POWER, &rilp,
```

Comment:
Please use ril_send_power() as ril_set_online(), so we do g_ril_send(RIL_REQUEST_RADIO_POWER) in only one place.